### PR TITLE
refactor(*): convert imports to @spinnaker/core

### PR DESCRIPTION
**Pull request by anotherchrisberry**
Moving all imports from the core module into a common alias, `@spinnaker/core`. Anything that is needed from core by a cloud provider should be exported via `core/index.ts`. If it's not available, it should be included there.

I've also, as much as possible, tried to standardize the import order/layout. I'll get a page together on the wiki documenting import order, which is basically:
* `angular` should always be the first thing `import`ed (TS, TSX) or `require`d as a `const` (JS)
* any third-party imports should follow `angular`
* a blank line
* a single import from `@spinnaker/core`, if applicable
* a blank line
* any imports from the current module
* a blank line
* any `.less` files, which should be `import`ed, not `require`d.

Other things we'd like to standardize on:
* `import`s should prefer module-relative importing to multiple `../`s.
* space between curly brackets on import statements, e.g. `{ module }` instead of `{module}`

It's not as if this is truly reviewable - it's 931 files - but @danielpeach if you could at least take a look at the Google modules and run through them, I'd appreciate that before merging this.

### CODE REVIEW

File: app/scripts/modules/amazon/image/image.reader.js [hunk 1]
```diff
@@ -1,8 +1,8 @@
 'use strict';
 
-import {API_SERVICE} from 'core/api/api.service';
+import { API_SERVICE } from '@spinnaker/core';
 
-let angular = require('angular');
+const angular = require('angular');
```
Comment (jrsquared) [inline app/scripts/modules/amazon/image/image.reader.js]: angular goes above internal imports

File: app/scripts/modules/amazon/keyPairs/keyPairs.read.service.js [hunk 2]
```diff
@@ -1,8 +1,8 @@
 'use strict';
 
-import {API_SERVICE} from 'core/api/api.service';
+import { API_SERVICE } from '@spinnaker/core';
 
-let angular = require('angular');
+const angular = require('angular');
```
Comment (jrsquared) [inline app/scripts/modules/amazon/keyPairs/keyPairs.read.service.js]: angular goes above internal imports

File: app/scripts/modules/amazon/loadBalancer/loadBalancer.transformer.js [hunk 3]
```diff
@@ -3,7 +3,7 @@
 import _ from 'lodash';
 import {AWSProviderSettings} from '../aws.settings';
 
-let angular = require('angular');
+const angular = require('angular');
```
Comment (jrsquared) [inline app/scripts/modules/amazon/loadBalancer/loadBalancer.transformer.js]: angular goes above with lodash

File: app/scripts/modules/amazon/loadBalancer/loadBalancer.transformer.js [hunk 4]
```diff
@@ -3,7 +3,7 @@
 import _ from 'lodash';
 import {AWSProviderSettings} from '../aws.settings';
```
Comment (jrsquared) [inline app/scripts/modules/amazon/loadBalancer/loadBalancer.transformer.js]: space

File: app/scripts/modules/amazon/pipeline/stages/destroyAsg/awsDestroyAsgStage.js [hunk 5]
```diff
@@ -1,7 +1,7 @@
 'use strict';
 
-let angular = require('angular');
-import {StageConstants} from 'core/pipeline/config/stages/stageConstants';
+const angular = require('angular');
+import { StageConstants } from '@spinnaker/core';
```
Comment (jrsquared) [inline app/scripts/modules/amazon/pipeline/stages/destroyAsg/awsDestroyAsgStage.js]: space

File: app/scripts/modules/amazon/pipeline/stages/disableAsg/awsDisableAsgStage.js [hunk 6]
```diff
@@ -1,7 +1,7 @@
 'use strict';
 
-let angular = require('angular');
-import {StageConstants} from 'core/pipeline/config/stages/stageConstants';
+const angular = require('angular');
+import { StageConstants } from '@spinnaker/core';
```
Comment (jrsquared) [inline app/scripts/modules/amazon/pipeline/stages/disableAsg/awsDisableAsgStage.js]: space

File: app/scripts/modules/amazon/pipeline/stages/modifyScalingProcess/modifyScalingProcessStage.js [hunk 7]
```diff
@@ -1,7 +1,7 @@
 'use strict';
 
-let angular = require('angular');
-import {StageConstants} from 'core/pipeline/config/stages/stageConstants';
+const angular = require('angular');
+import { StageConstants } from '@spinnaker/core';
```
Comment (jrsquared) [inline app/scripts/modules/amazon/pipeline/stages/modifyScalingProcess/modifyScalingProcessStage.js]: space

File: app/scripts/modules/amazon/securityGroup/configure/configSecurityGroup.mixin.controller.js [hunk 8]
```diff
@@ -1,15 +1,17 @@
...
-import {Subject} from 'rxjs';
-
-import {ACCOUNT_SERVICE} from 'core/account/account.service';
-import {V2_MODAL_WIZARD_SERVICE} from 'core/modal/wizard/v2modalWizard.service';
-import {SECURITY_GROUP_READER} from 'core/securityGroup/securityGroupReader.service';
-import {SECURITY_GROUP_WRITER} from 'core/securityGroup/securityGroupWriter.service';
-import {TASK_MONITOR_BUILDER} from 'core/task/monitor/taskMonitor.builder';
-import {AWSProviderSettings} from '../../aws.settings';
+import { Subject } from 'rxjs';
+import { AWSProviderSettings } from 'amazon/aws.settings';
```
Comment (jrsquared) [inline app/scripts/modules/amazon/securityGroup/configure/configSecurityGroup.mixin.controller.js]: move down with non-external imports

File: app/scripts/modules/amazon/serverGroup/configure/serverGroupCommandBuilder.service.js [hunk 9]
```diff
@@ -1,14 +1,10 @@
 'use strict';
 
+const angular = require('angular');
 import _ from 'lodash';
+import { AWSProviderSettings } from 'amazon/aws.settings';
```
Comment (jrsquared) [inline app/scripts/modules/amazon/serverGroup/configure/serverGroupCommandBuilder.service.js]: move down with non-external imports

File: app/scripts/modules/amazon/loadBalancer/configure/createLoadBalancer.controller.js [hunk 10]
```diff
@@ -1,20 +1,21 @@
...
+  CACHE_INITIALIZER_SERVICE,
+  INFRASTRUCTURE_CACHE_SERVICE,
+  LOAD_BALANCER_WRITE_SERVICE,
+  NAMING_SERVICE,
+  SECURITY_GROUP_READER,
+  SUBNET_READ_SERVICE,
+  TASK_MONITOR_BUILDER,
+  V2_MODAL_WIZARD_SERVICE
+} from '@spinnaker/core';
+import { AWSProviderSettings } from '../../aws.settings';
```
Comment (jrsquared) [inline app/scripts/modules/amazon/loadBalancer/configure/createLoadBalancer.controller.js]: can these`../../` just be `amazon/`?

File: app/scripts/modules/appengine/appengine.module.ts [hunk 11]
```diff
@@ -1,21 +1,23 @@
...
+import { APPENGINE_INSTANCE_DETAILS_CTRL } from './instance/details/details.controller';
+import { APPENGINE_LOAD_BALANCER_MODULE } from './loadBalancer/loadBalancer.module';
+import { APPENGINE_PIPELINE_MODULE } from './pipeline/pipeline.module';
+import { APPENGINE_SERVER_GROUP_COMMAND_BUILDER } from './serverGroup/configure/serverGroupCommandBuilder.service';
+import { APPENGINE_SERVER_GROUP_BASIC_SETTINGS_CTRL } from './serverGroup/configure/wizard/basicSettings.controller';
+import { APPENGINE_CLONE_SERVER_GROUP_CTRL } from './serverGroup/configure/wizard/cloneServerGroup.controller';
+import { APPENGINE_SERVER_GROUP_DETAILS_CTRL } from './serverGroup/details/details.controller';
+import { APPENGINE_SERVER_GROUP_TRANSFORMER } from './serverGroup/transformer';
+import { APPENGINE_SERVER_GROUP_WRITER } from './serverGroup/writer/serverGroup.write.service';
+
```
Comment (jrsquared) [inline app/scripts/modules/appengine/appengine.module.ts]: why this space?

File: app/scripts/modules/appengine/loadBalancer/configure/wizard/stageAllocationConfigurationRow.component.ts [hunk 12]
```diff
@@ -1,8 +1,8 @@
-import {module} from 'angular';
-import {uniq} from 'lodash';
-import {StageConstants} from 'core/pipeline/config/stages/stageConstants';
-import {IAppengineAllocationDescription} from 'appengine/loadBalancer/transformer';
-import {Application} from 'core/application/application.model';
+import { Application, StageConstants } from '@spinnaker/core';
```
Comment (jrsquared) [inline app/scripts/modules/appengine/loadBalancer/configure/wizard/stageAllocationConfigurationRow.component.ts]: move down to internal imports

**Comment by anotherchrisberry** [comment]
There will definitely be some issues with this, but we'll try to address them quickly as they come up...

**Pull request merged**